### PR TITLE
Update test_pipeline.py for add two cases

### PR DESCRIPTION
**Pull request by kychennv**

### CODE REVIEW

**Comment by jantonguirao** [comment]
Build 690882

File: dali/test/python/test_pipeline.py [hunk 1]
```diff
@@ -662,6 +662,146 @@ def iter_setup(self):
...
+            pass
+    
+    nonfused_pipe = NonFusedPipeline(batch_size=batch_size, num_threads=1, device_id = 0, num_gpus = 1)
+    nonfused_pipe.build()
+    nonfused_pipe_out = nonfused_pipe.run()
+    fused_pipe = FusedPipeline(batch_size=batch_size, num_threads=1, device_id = 0, num_gpus = 1)
+    fused_pipe.build()
+    fused_pipe_out = fused_pipe.run()
+    for i in range(batch_size):
+        nonfused_pipe_out_cpu = nonfFused_pipe_out[0].as_cpu()
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]: This fails:
```
======================================================================
ERROR: Comparing results of pipeline: (nvJPEGDecoder -> RandomCrop), with the same operation performed by fused operator
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/opt/dali/dali/test/python/test_pipeline.py", line 735, in test_equal_nvJPEGDecoderRandomCrop_nvJPEGDecoder
    nonfused_pipe_out_cpu = nonfFused_pipe_out[0].as_cpu()
NameError: global name 'nonfFused_pipe_out' is not defined

======================================================================
ERROR: Comparing results of pipeline: (nvJPEGDecoder -> Slice), with the same operation performed by fused operator
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/opt/dali/dali/test/python/test_pipeline.py", line 796, in test_equal_nvJPEGDecoderSlice_nvJPEGDecoder
    nonfused_pipe_out = nonfused_pipe.run()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 314, in run
    self._run()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 320, in _run
    self._prefetch()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 332, in _prefetch
    self._run_once()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 343, in _run_once
    self.iter_setup()
  File "/opt/dali/dali/test/python/test_pipeline.py", line 767, in iter_setup
    (crop_pos, crop_size) = pos_size_iter.next()
NameError: global name 'pos_size_iter' is not defined

----------------------------------------------------------------------
```

Comment (JanuszL) [inline dali/test/python/test_pipeline.py]:
```suggestion
        nonfused_pipe_out_cpu = nonfused_pipe_out[0].as_cpu()
```
?

**Comment by jantonguirao** [comment]
Build 692247

**Comment by jantonguirao** [comment]
Build 692337

**Comment by jantonguirao** [comment]
Build 693854

**Comment by jantonguirao** [comment]
@kychennv There is another error:
```
======================================================================
ERROR: Comparing results of pipeline: (nvJPEGDecoder -> Slice), with the same operation performed by fused operator
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/opt/dali/dali/test/python/test_pipeline.py", line 796, in test_equal_nvJPEGDecoderSlice_nvJPEGDecoder
    nonfused_pipe_out = nonfused_pipe.run()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 314, in run
    self._run()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 320, in _run
    self._prefetch()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 332, in _prefetch
    self._run_once()
  File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 343, in _run_once
    self.iter_setup()
  File "/opt/dali/dali/test/python/test_pipeline.py", line 767, in iter_setup
    (crop_pos, crop_size) = pos_size_iter.next()
NameError: global name 'pos_size_iter' is not defined

======================================================================
FAIL: Comparing results of pipeline: (nvJPEGDecoder -> RandomCrop), with the same operation performed by fused operator
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/opt/dali/dali/test/python/test_pipeline.py", line 737, in test_equal_nvJPEGDecoderRandomCrop_nvJPEGDecoder
    assert(np.sum(np.abs(nonfused_pipe_out_cpu.at(i)-fused_pipe_out_cpu.at(i)))==0)
AssertionError
```

Please run the tests locally before pushing

**Comment by kychennv** [comment]
> @kychennv There is another error:
> 
> ```
> ======================================================================
> ERROR: Comparing results of pipeline: (nvJPEGDecoder -> Slice), with the same operation performed by fused operator
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/opt/dali/dali/test/python/test_pipeline.py", line 796, in test_equal_nvJPEGDecoderSlice_nvJPEGDecoder
>     nonfused_pipe_out = nonfused_pipe.run()
>   File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 314, in run
>     self._run()
>   File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 320, in _run
>     self._prefetch()
>   File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 332, in _prefetch
>     self._run_once()
>   File "/usr/local/lib/python2.7/dist-packages/nvidia/dali/pipeline.py", line 343, in _run_once
>     self.iter_setup()
>   File "/opt/dali/dali/test/python/test_pipeline.py", line 767, in iter_setup
>     (crop_pos, crop_size) = pos_size_iter.next()
> NameError: global name 'pos_size_iter' is not defined
> 
> ======================================================================
> FAIL: Comparing results of pipeline: (nvJPEGDecoder -> RandomCrop), with the same operation performed by fused operator
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
>     self.test(*self.arg)
>   File "/opt/dali/dali/test/python/test_pipeline.py", line 737, in test_equal_nvJPEGDecoderRandomCrop_nvJPEGDecoder
>     assert(np.sum(np.abs(nonfused_pipe_out_cpu.at(i)-fused_pipe_out_cpu.at(i)))==0)
> AssertionError
> ```
> Please run the tests locally before pushing

HI, i have fixed the slice code. but for the random, may be as same issue as 200505129

File: dali/test/python/test_pipeline.py [hunk 2]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+        def iter_setup(self):
+            pass
+
+    class FusedPipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(FusedPipeline, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id,seed=123456)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.decode = ops.nvJPEGDecoderRandomCrop(device = "mixed", output_type = types.RGB)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
            self.decode = ops.nvJPEGDecoderRandomCrop(device = "mixed", output_type = types.RGB, seed=self.seed)
```

File: dali/test/python/test_pipeline.py [hunk 3]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+
+        def iter_setup(self):
+            pass
+
+    class FusedPipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(FusedPipeline, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id,seed=123456)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
            self.seed = seed
            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
```

File: dali/test/python/test_pipeline.py [hunk 4]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+        Comparing results of pipeline: (nvJPEGDecoder -> RandomCrop), with the same operation performed by fused operator
+    """
+    batch_size =128
+
+    class NonFusedPipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(NonFusedPipeline, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id,seed=123456)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
            self.seed = seed
            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
```

File: dali/test/python/test_pipeline.py [hunk 5]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+            self.decode = ops.nvJPEGDecoder(device = "mixed", output_type = types.RGB)
+            self.res = ops.RandomResizedCrop(device="gpu", size =(224,224))
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
            self.res = ops.RandomResizedCrop(device="gpu", size =(224,224), seed=self.seed)
```

File: dali/test/python/test_pipeline.py [hunk 6]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
         assert(np.sum(np.abs(nonfused_pipe_out_cpu.at(i)-fused_pipe_out_cpu.at(i)))==0)
 
+def test_equal_nvJPEGDecoderRandomCrop_nvJPEGDecoder():
+    """
+        Comparing results of pipeline: (nvJPEGDecoder -> RandomCrop), with the same operation performed by fused operator
+    """
+    batch_size =128
+
+    class NonFusedPipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
        def __init__(self, batch_size, num_threads, device_id, num_gpus, seed=123456):
```

File: dali/test/python/test_pipeline.py [hunk 7]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+            super(NonFusedPipeline, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id,seed=123456)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
                                             device_id)
```

File: dali/test/python/test_pipeline.py [hunk 8]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+
+
+        def iter_setup(self):
+            pass
+
+    class FusedPipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(FusedPipeline, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id,seed=123456)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
                                             device_id)
```

File: dali/test/python/test_pipeline.py [hunk 9]
```diff
@@ -662,6 +662,168 @@ def iter_setup(self):
...
+            mirror = self.coin()
+            output = self.cmnp(resized_images, mirror = mirror, crop_pos_x = self.uniform(), crop_pos_y = self.uniform())
+            return (output,resized_images, self.labels)
+
+
+        def iter_setup(self):
+            pass
+
+    class FusedPipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]:
```suggestion
        def __init__(self, batch_size, num_threads, device_id, num_gpus, seed=123456):
```

**Comment by jantonguirao** [comment]
Build 695144

**Comment by jantonguirao** [comment]
Build 695241

**Comment by jantonguirao** [comment]
Build 695246

**Pull request merged**